cloudsupport/cloudproviderconfiguration.go

```diff
@@ -97,3 +98,29 @@ func GetDescriptiveInfoFromCloudProvider(cluster string, cloudProvider string) (
 
 	return clusterInfo, nil
 }
+
+// GetDescribeRepositoriesFromCloudProvider returns image repository descriptions from the cloud provider wrapped in IMetadata obj
+func GetDescribeRepositoriesFromCloudProvider(cluster string, cloudProvider string) (workloadinterface.IMetadata, error) {
+	var clusterInfo *cloudsupportv1.CloudProviderDescribeRepositories
+
+	switch cloudProvider {
+	case cloudsupportv1.EKS:
+		eksSupport := cloudsupportv1.NewEKSSupport()
+		region, err := eksSupport.GetRegion(cluster)
+		if err != nil {
+			return nil, err
+		}
+		clusterInfo, err = cloudsupportv1.GetDescribeRepositoriesEKS(eksSupport, cluster, region)
+		if err != nil {
+			return nil, err
+		}
+	case cloudsupportv1.GKE:
```

Comment:
[Question]: Should we return here an error indicating that it is not implemented?

Reply:
I was debating whether to do that or check in KS we received an obj or nil. What do you think?

Reply:
currently KS assumes that is it gets an error for this pkg it's because of an integration problem. (and will show a control is skipped for that reason). In this case that would be wrong.
Do you think KS should handle the different types of errors or leave like this and check if received an object/nil?

Reply:
mmm It's a good question. 
The correct approach is to create different error objects, so if we had an error with integration we will return the integration error and if it is not implemented we will return a different error.
The package calling this function will handle the different errors.
I recommend you read about it, but it's not a blocker.